pkg/util/extract_forwarded_test.go

```diff
@@ -0,0 +1,59 @@
+package util
+
+import (
+	"context"
+	"testing"
+
+	"google.golang.org/grpc/metadata"
+)
+
+func TestGetSourceFromOutgoingCtx(t *testing.T) {
+	tests := []struct {
+		name  string
+		key   string
+		value string
+		want  string
+	}{
+		{
+			name:  "No value in key",
+			key:   ipAddressesKey,
+			value: "",
+			want:  "",
+		},
+		{
+			name:  "Value in key",
+			key:   ipAddressesKey,
+			value: "172.16.1.1",
+			want:  "172.16.1.1",
+		},
+		{
+			name:  "Stored under wrong key",
+			key:   "wrongkey",
+			value: "172.16.1.1",
+			want:  "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Test extracting from incoming context
+			ctx := context.Background()
+			if tt.value != "" {
+				md := metadata.Pairs(tt.key, tt.value)
+				ctx = metadata.NewIncomingContext(ctx, md)
+			}
+			if got := GetSourceFromIncomingCtx(ctx); got != tt.want {
+				t.Errorf("GetSourceFromOutgoingCtx() = %v, want %v", got, tt.want)
```

Comment:
We use the `testify` framework in Cortex. You can simplify the assertions in this file using `assert.Equal(t, want, got)`.

Reply:
Changed.